.github/workflows/mariadb.yml

```
@@ -0,0 +1,67 @@
+name: Build Mroonga with MariaDB main
+on:
+  push:
+  pull_request:
+    paths:
+      - "**/*.cpp"
+      - "**/*.h"
+      - "**/*.hpp"
+      - "**/CMakeLists.txt"
+      - "**/Makefile.am"
+      - ".github/workflows/mariadb.yml"
+      - "configure.ac"
+      - "doc/**"
+      - "mysql-test/**"
+      - "packages/**"
+      - "version_*"
+  schedule:
+    - cron: |
+        0 0 * * *
+concurrency:
+  group: ${{ github.head_ref || github.sha }}-${{ github.workflow }}
+  cancel-in-progress: true
+jobs:
+  build:
+    name: Build with MariaDB main
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ruby
+      - name: Install required packages
+        run: |
+          sudo apt install -y \
+            build-essential \
+            ninja-build
+      - uses: actions/checkout@v4
+        with:
+          path: mariadb
+          repository: MariaDB/server
+      - name: Remove bundled Mroonga
+        run: |
+          cd mariadb
+          git rm -r --cached storage/mroonga/
+          rm -rf storage/mroonga
+      - uses: actions/checkout@v4
+        with:
+          path: mariadb/extra/groonga
+          repository: groonga/groonga
+          submodules: true
+      - uses: actions/checkout@v4
+        with:
+          path: mariadb/storage/mroonga
+          repository: mroonga/mroonga
+          submodules: true
+      - name: Build MariaDB main
+        run: |
+          mkdir build-mariadb
+          cmake \
+            -S mariadb \
+            -B build-mariadb \
+            -GNinja \
```

[nits]
ドキュメントのビルド時の命名とあっていると統一感があってよいなぁとおもいました。
https://github.com/mroonga/mroonga/blob/main/doc/source/developer/debug.md#how-to-build-mysql-for-debugging
```suggestion
            -Smariadb \
            -Bmariadb-main.build \
            -GNinja \
```